source/guide_teamcity.rst

```diff
@@ -0,0 +1,162 @@
+.. highlight:: console
+
+.. author:: Direnc <direnc99@gmail.com>
+
+.. categorize your guide! refer to the current list of tags: https://lab.uberspace.de/tags
+.. tag:: web
+.. tag:: self-hosting
+
+
+.. sidebar:: About
+
+  .. image:: _static/images/teamcity.svg
+      :align: center
+      :target: https://www.jetbrains.com/company/brand/logos/
+
+
+##########
+TeamCity
+##########
+
+.. tag_list::
+
+TeamCity_ is a build management tool by JetBrains. It is not open source and requires a license, though it offers a freemium model with up to 100 build configurations and 3 build agents. TeamCity makes it possible to attach a server to your projects in different Version Control Systems so that it runs build steps on pre-configured triggers.
+
+----
+
+.. note:: For this guide you should be familiar with the basic concepts of
+
+  * :manual:`MySQL <database-mysql>`
+  * :manual:`domains <web-domains>`
+
+License
+=======
+
+All relevant legal information can be found here
+
+  * https://www.jetbrains.com/teamcity/buy/license.html
+
+Logo source: https://www.jetbrains.com/company/brand/logos/
+
+Prerequisites
+=============
+
+We're using Java in the stable version 14.0.2:
+
+::
+
+ [isabell@stardust ~]$ java --version
+ openjdk 14.0.2 2020-07-14
+ OpenJDK Runtime Environment 20.3 (build 14.0.2+12)
+ OpenJDK 64-Bit Server VM 20.3 (build 14.0.2+12, mixed mode, sharing)
+ [isabell@stardust ~]$
+
+MySQL
+-----
+
+.. include:: includes/my-print-defaults.rst
+
+We will create a separate database for TeamCity:
+
+::
+
+ [isabell@stardust ~]$ mysql -e "CREATE DATABASE ${USER}_teamcity"
+ [isabell@stardust ~]$
+
+Your URL needs to be setup:
+
+.. include:: includes/web-domain-list.rst
+
+You can also setup your own :manual_anchor:`domain <web-domains.html>`. TeamCity will run as a backend serving HTML files.
+
+Installation
+============
+
+Step 1
+------
+First, we're going to download TeamCity and extract it.
+
+Change into your local bin folder. Then we're going to download the latest version (TeamCity-Download_ page) of the tgz archive and extract it. Note that the download link on the page is actually a redirect. So instead of the link behind the button, use the direct link e.g.: ``https://download.jetbrains.com/teamcity/TeamCity-2020.1.5.tar.gz``.
+
+::
+
+ [isabell@stardust ~]$ cd ~/bin
+ [isabell@stardust bin]$ wget -O teamcity.tgz https://download.jetbrains.com/teamcity/TeamCity-2020.1.5.tar.gz
+ [isabell@stardust bin]$ tar --gzip --extract --file=teamcity.tgz
+ [isabell@stardust bin]$ rm teamcity.tgz
+ [isabell@stardust bin]$
+
+Step 2
+------
+Start the TeamCity Server. The startup scripts lie in TeamCity/bin
+
+::
+
+ [isabell@stardust bin]$ ./TeamCity/bin/runAll.sh start
```

Comment:
All services on uberspace should be started within supervisord. This ensures that they stay online, if a server gets a reboot or is having temporary problems.